### DISCUSSION TITLE
Decontx clusterstering looks weird for libraries which might affect analysis

### DISCUSSION THREAD

**Discussion by lquin003** [campbio/celda#383] [General]
Hello, I know that Decontx uses cluster information to estimate the amount of contamination outside of those clusters and therefore the clusters that decontx creates are important. My concern is that the clusters that decontx creates using the single cell experiment assay object looks nothing like my actual clusters after processing. This might suggest that the estimates are incorrect based on the variability in clusters. Is this something that others have seen in the past? If so how did you go about correcting this or using this tool?

**This plot comes from seurat after QC filtering for features/MT reads**
![image](https://user-images.githubusercontent.com/54153678/202336429-d57b639c-404a-4bcf-8e23-c1fda6f0e750.png)

**This plot comes from Single Cell Experiment/Decontx make UMAP function before QC (Notice how most cells fall in cluster 1)**
![image](https://user-images.githubusercontent.com/54153678/202336559-1840eba0-b71c-4451-a663-463aeeacad82.png)

**Comment by joshua-d-campbell** [comment]
Hi @lquin003, that does look quite strange. Can you include the code you used to run decontX as well as the output from `sessionInfo()` so I can see the version you are using? 

In the meantime, you can input your set of Seurat cluster labels using the `z` parameter and decontX will use those.

**Comment by lquin003** [comment]
Hi Joshua,

I figured out the problem. The matrix was transposed thanks for your response.

Best,
Luis

**Comment by joshua-d-campbell** [comment]
Thanks for updating the thread! Also, you may want to use a newer versions of the celda package as there are some tweaks when it comes to error checking.
